File: MediafyDemoAppJava/app/src/main/java/com/mediafy/demo/java/activities/BaseAdActivity.java


/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */

package com.mediafy.demo.java.activities;

import android.os.Bundle;
import android.view.ViewGroup;
import androidx.annotation.CallSuper;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import com.mediafy.demo.java.R;
import com.mediafy.demo.java.databinding.ActivityDemoBinding;
import com.mediafy.demo.java.testcases.TestCase;
import com.mediafy.demo.java.testcases.TestCaseRepository;

public class BaseAdActivity extends AppCompatActivity {

    protected static final String TAG = "AdExampleActivity";
    protected ActivityDemoBinding binding;

    private TestCase testCase = TestCaseRepository.lastTestCase;

    @Override
    @CallSuper
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_demo);
        binding.tvTestCaseName.setText(testCase.getName());
    }

    protected ViewGroup getContainerForAd() {
        return binding.frameAdWrapper;
    }

}

File: MediafyDemoAppKotlin/app/src/main/java/com/mediafy/demo/kotlin/activities/BaseAdActivity.kt
/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */



package com.mediafy.demo.kotlin.activities

import android.os.Bundle
import android.view.ViewGroup
import androidx.annotation.CallSuper
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.mediafy.demo.kotlin.R
import com.mediafy.demo.kotlin.databinding.ActivityDemoBinding
import com.mediafy.demo.kotlin.testcases.TestCase
import com.mediafy.demo.kotlin.testcases.TestCaseRepository

open class BaseAdActivity : AppCompatActivity() {

    companion object {
        const val TAG = "AdExampleActivity"
    }

    /**
     * ViewGroup container for any ad view.
     */
    protected val containerForAd: ViewGroup
        get() = binding.frameAdWrapper

    private lateinit var binding: ActivityDemoBinding
    private var testCase: TestCase = TestCaseRepository.lastTestCase

    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_demo)
        binding.tvTestCaseName.text = testCase.name
    }

}